src/applicationcomposer/messageHandlers/deployMessageHandler.ts

```diff
@@ -3,10 +3,49 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import vscode from 'vscode'
+import vscode, { Uri } from 'vscode'
 import { Command, DeployRequestMessage, DeployResponseMessage, MessageType, WebviewContext } from '../types'
 import { SamSyncResult } from '../../shared/sam/sync'
 import { telemetry } from '../../shared/telemetry/telemetry'
+import { Auth } from '../../auth/auth'
+import { promptAndUseConnection } from '../../auth/utils'
+import { openUrl } from '../../shared/utilities/vsCodeUtils'
+import { localize } from 'vscode-nls'
+import { StatefulConnection } from '../../auth/connection'
+
+const credentialsDocLink = 'https://docs.aws.amazon.com/toolkit-for-vscode/latest/userguide/setup-credentials.html'
+
+async function promptReauth(connection?: StatefulConnection) {
+    let errorMessage = localize(
+        'aws.applicationComposer.deploy.authModal.message',
+        'Syncing requires authentication with IAM credentials.'
+    )
+    if (connection?.state === 'valid') {
+        errorMessage =
+            localize(
+                'aws.applicationComposer.deploy.authModal.invalidAuth',
+                'Authentication through Builder ID or IAM Identity Center detected. '
+            ) + errorMessage
+    }
+    const acceptMessage = localize(
+        'aws.applicationComposer.deploy.authModal.accept',
+        'Authenticate with IAM credentials'
+    )
+    const docMessage = localize('aws.applicationComposer.deploy.authModal.docLink', 'Open documentation')
```

Comment:
not needed when using `showMessageWithUrl `

Reply:
Our docs writer mentioned that "Open documentation" is preferred over "View documentation". Should this change be made across the entire Toolkit?

Reply:
> Should this change be made across the entire Toolkit?

Not sure, but that should be a separate PR. The Toolkit should use consistent language.